src/core.jl

```diff
@@ -383,7 +383,7 @@ Composition of transducers.
 @inline Base.:∘(g::Transducer, f::Composition) = g ∘ f.inner ∘ f.outer
 @inline Base.:∘(f::Transducer, ::IdentityTransducer) = f
 @inline Base.:∘(::IdentityTransducer, f::Transducer) = f
-@inline Base.:∘(::IdentityTransducer, ::IdentityTransducer) = IdentityTransducer()
+@inline Base.:∘(f::IdentityTransducer, ::IdentityTransducer) = f
```

Comment:
I am curious, why you did this change?

Reply:
Totally no reason :sweat_smile:. I just wrote it this way in #323 and carried it over while solving the merge conflict (a bit hoping that it'd be close enough for the diff viewer to pretty-print it).

Reply:
Thanks for the clarification. I thought maybe this works around some compiler quirk that I should know about.